backend/src/indexer/block_preprocessor.rs

```diff
@@ -294,21 +298,53 @@ impl concordium_rust_sdk::indexer::Indexer for BlockPreProcessor {
 pub async fn compute_total_stake_capital(
     client: &mut v2::Client,
     block_height: v2::BlockIdentifier,
-) -> v2::QueryResult<Amount> {
+) -> v2::QueryResult<(Amount, HashMap<i64, ValidatorStakingInformation>)> {
```

Comment:
Since we are only inserting and then traversing and not really using the properties of a hashmap, a `Vec<(i64, ValidatorStakingInformatio)>` would mean less overhead.
If you go for the single SQL statement to insert it should probably be several `Vec`s